File: 0053-maximum-subarray/0053-maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        ele_sum , max_sum = 0 , float('-inf')
        
        for num in nums :
            ele_sum += num
            
            if max_sum < ele_sum :
                max_sum = ele_sum
            
            if ele_sum < 0 :
                ele_sum = 0
        
        return max_sum

File: 0053-maximum-subarray/0053-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int max_sum = Integer.MIN_VALUE;
        int temp = 0;
        for(int i=0;i<nums.length;i++){
            temp += nums[i];
            max_sum = Math.max(max_sum,temp);
            if(temp < 0)
                temp = 0;
        }
        return max_sum;
    }
}